1-JavaScript/Week3/homework/ex6-totalCost.js

```diff
@@ -21,11 +21,18 @@ instead!
 -----------------------------------------------------------------------------*/
 const cartForParty = {
   // TODO complete this function
+  chocolate: 2.99, chips: 1.99, tea: 1, milk: 0.99, eggs: 4
 };
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
+function calculateTotalPrice(cart) {
   // TODO replace this comment with your code
+  let total = 0;
+  for (const price in cart){
+      total += cart[price];
+  }
+ return `Total: €${total}`
```

Comment:
Hi Rana - although this logic works fine but you could improve it slightly. Firstly, you should try to avoid using `for-in` loops for iterating over objects. Try to use `Object.keys()` instead. The difference between the two can be found out on google, it's something to do regarding iterating over parent object properties. Secondly, if you use Object.keys, you could also look at using the array's reduce method. Something like below:

```
function calculateTotalPrice(cart) {
  const total = Object.keys(cart).reduce((acc, elem) => acc + cart[elem], 0);
  return `Total: ${total}`;
}
```